src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/AddPackageCommandUtility.cs

```diff
@@ -116,5 +117,19 @@ public static List<PackageSource> EvaluateSources(IList<PackageSource> requested
                 return packageSources;
             }
         }
+
+        /// <summary>
+        /// Updates package sources
+        /// </summary>
+        /// <param name="packageSpec">Package spec which may need update source feed</param>
+        /// <param name="newSources">New user supplied source feeds as argument</param>
+        public static void UpdateSourceFeeds(PackageSpec packageSpec, string[] newSources)
+        {
+            if (newSources?.Any() == true)
+            {
+                packageSpec.RestoreMetadata.Sources =
+                    newSources.Where(ns => !string.IsNullOrEmpty(ns)).ToHashSet(StringComparer.OrdinalIgnoreCase).Select(ns => new PackageSource(ns)).ToList();
```

Comment:
Do we need the hashset? 
Seems like an overkill given that the rest of the code likely handles it and it's an uncommon scenario. 

Or did you already hit a problem with dup sources and my assumption is inaccurate.

Reply:
Removed hashset.